### PR TITLE
update gradle to 6.8

### PR DESCRIPTION
**Pull request by dsvoronin**
https://docs.gradle.org/6.8/release-notes.html

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -6,7 +6,7 @@ org.gradle.caching=true
 android.useAndroidX=true
 # Current stable version to be used in this project
 infraVersion=2020.36
-systemProp.kotlinVersion=1.3.72
+systemProp.kotlinVersion=1.4.21
```
Comment (eugene-krivobokov) [inline gradle.properties:9]: Ideally, it can be done in a different PR

File: subprojects/gradle/buildchecks/build.gradle.kts [hunk 2]
```diff
@@ -18,7 +18,6 @@ dependencies {
     implementation(project(":gradle:kotlin-dsl-support"))
     implementation(Dependencies.Gradle.kotlinPlugin)
     implementation(Dependencies.funktionaleTry)
-    implementation(Dependencies.Gradle.androidPlugin)
```
Comment (eugene-krivobokov) [inline subprojects/gradle/buildchecks/build.gradle.kts:21]: How did you find this unnecessary dependency? 
The `dependency-analysis` plugin?

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt [hunk 3]
```diff
@@ -229,16 +229,17 @@ abstract class InstrumentationTestsTask @Inject constructor(
 
     private fun createReportFactory(): Report.Factory {
         val reportViewerConfig = reportViewerConfig.orNull
-        val factories = mutableMapOf<Class<out Report.Factory.Config>, Report.Factory>()
+        val factories = mutableMapOf<String, Report.Factory>()
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt:232]: Why did you weaken a key? A string is more error-prone, easier to miss.

File: subprojects/gradle/buildchecks/build.gradle.kts (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/buildchecks/build.gradle.kts:21]: nope, manual check

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt:232]: New type inference in 1.4 breaks this and i don't know how to fix it properly

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**